docs/extend/mail.md

```diff
@@ -0,0 +1,58 @@
+# Mail
+
+In addition to the [default drivers in core](../mail.md), Flarum allows new mail drivers to be added through extenders. To create your own mail driver, you'll need to create a class implementing `\Flarum\Mail\DriverInterface`. Flarum actually takes care of the frontend for providing email settings: just declare which settings you need, and any default values, in `availableSettings`.
+
+For example:
+
+```php
+use Flarum\Mail\DriverInterface;
+use Illuminate\Contracts\Validation\Factory;
+use Illuminate\Mail\Transport\MailgunTransport;
+use Illuminate\Support\MessageBag;
+use Swift_Transport;
+
+class MailgunDriver implements DriverInterface
+{
+    public function availableSettings(): array
+    {
+        return [
+            'setting_one' => '',
+            'setting_two' => 'defaultValue',
+            'dropdown_setting' => [
+                'option_one_val' => 'Option One Display',
```

Comment:
I don't remember if those could be translated? If so we should show how to have them translated.

Reply:
I don't think so unfortunately, although we should add that...